sips/sip-011/sip-011-catastrophic-failure-recovery.md

```diff
@@ -0,0 +1,464 @@
+  mining transactions that interact with it on the canonical fork.  For example,
+a soft fork can be used to prevent miners from allowing an attacker to withdraw
+digital assets from a popular but buggy smart contract.
+* Preventing or constraining the blockchain from mining particular types of
+  transactions that may be mined on the canonical fork.  For example, if a
+partial cryptographic break is discovered for the Stacks blockchain's
+key-signing algorithm that enables hackers to calculate some private keys from
+on-chain data, a soft fork could prevent any transactions signed with vulnerable
+keys from being mined until the true owner can authenticate themselves to the
+miners through some other channel.
+* Declaring a "flag day" for a subsequent hard fork, by declaring that the
+  current node software will reach end-of-life status by a certain burnchain
+block height.  This may be appropriate if the bugfix for the catastrophic
+failure is temporary (e.g. if the design of concensus-critical code-paths in the
+blockchain are at fault, and preclude a permanent fix).
+
+There are multiple different ways to execute a soft fork.  This document only
+discusses procedures for a _miner-actived_ soft fork whose goal is to address a
+liveness or safety failure.  A subsequent SIP may codify other types of soft
+forks and their applications in different contexts.  Due to the comparatively
+low barrier-to-entry for mining on the Stacks chain, miner-activated soft forks
+are expected to offer an acceptable measurement of the ecosystem's sentiments on
+a pending soft fork.
+
+A miner-activated soft fork procedure for healing a catastrophic failure works
+as follows:
+
+1. The Foundation and at least one other significant ecosystem entity will approve and
+sign a Stacks node software release that implements the soft fork rules.  This
+is the first step in this process because this new node software must address
+whatever catastrophic bug that led to the need to follow this procedure.  The
+remainder of this procedure is focused on codifying the new rules via the SIP
+process in order to require them in future node software releases.  The act of
+releasing this new node software will follow the critical bugfix point-release
+procedure, but with an additional warning from the announcement email that this
+new release is the beginning of a soft-fork roll-out.
+2. Once the catastrophic bug has been addressed, and a reliable majority of
+blocks on the canonical fork are produced by miners running the new software, a
+SIP describing the soft fork will be written and published describing the new
+rules and why they are necessary.  This SIP must then be advanced in the SIP
+process to "activation-in-progress" status by the relevant advisory boards and
+technical steering committee.  Among other things, this SIP's activation
+criteria must require the following:
+
+   * At least 80% of all miners must _explicitly and continuously vote to
```

Comment:
How do we measure this? Is this 80% of nodes on the network that have "miner=true"?

Reply:
80% of all blocks produced within the time interval.  The set of miners (really, their mining power) is fungible.  Will clarify.

Reply:
Addressed in 3832ae2